src/sentry/monitors/system_incidents.py

```diff
@@ -166,6 +177,258 @@ def get_clock_tick_volume_metric(tick: datetime) -> float | None:
+    #
+    # If we're not already in an incident and the most recent metric decision
+    # is an incident, mark this tick as an incident and backfill all abnormal
+    # decisions to an incident decision.
+    if last_decision != Decision.INCIDENT and metric_decisions[-1] == Decision.INCIDENT:
+        _backfill_decisions(past_ts, Decision.INCIDENT, Decision.ABNORMAL)
+        return make_decision(Decision.INCIDENT, AnomalyTransition.INCIDENT_STARTED)
+
+    # No decision transition. Use the previous decision
+    return make_decision(last_decision)
+
+
+def get_clock_tick_decision(tick: datetime) -> TickAnomalyDecision | None:
+    """
+    Retrieve the TickAnomalyDecision for a specific clock tick.
+    """
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+
+    if value := redis_client.get(MONITOR_TICK_DECISION.format(ts=_make_reference_ts(tick))):
+        return TickAnomalyDecision.from_str(value)
+    else:
+        return None
+
+
+def _backfill_decisions(
+    start: datetime,
+    decision: TickAnomalyDecision,
+    until_not: TickAnomalyDecision,
+) -> None:
+    """
+    Update historic tick decisions from `start` to `decision` until we no
+    longer see the `until_not` decision.
+    """
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+
+    ts = start
+    updates: dict[str | bytes, str] = {}
+
+    while True:
+        key = MONITOR_TICK_DECISION.format(ts=_make_reference_ts(ts))
+
+        # Nothing to backfill if we don't have a decision value
+        value = redis_client.get(key)
+        if value is None:
+            break
+
+        # Exit the backfill once we no longer have the until_not decision
+        prev_decision = TickAnomalyDecision.from_str(value)
+        if prev_decision != until_not:
+            break
+
+        updates[key] = decision.value
+        ts = ts - timedelta(minutes=1)
+
+    # Apply decision updates
+    if updates:
+        redis_client.mset(updates)
+
+
+def _metric_decision(pct_deviation: float | None) -> TickAnomalyDecision:
+    """
+    Determine an individual decision for the percentage deviation metric of a
+    clock tick. This only considers metrics that are negative, indicating
+    there's been a drop in check-in volume.
+    """
+    # examples: -5% anomaly and -25% incident
+    anomaly_threshold = options.get("crons.system_incidents.pct_deviation_anomaly_threshold")
+    incident_threshold = options.get("crons.system_incidents.pct_deviation_incident_threshold")
+
+    # If we do not have a metric for this tick we must assume things are
+    # operating normally
+    if pct_deviation is None:
+        return TickAnomalyDecision.NORMAL
+
+    if pct_deviation > anomaly_threshold:
+        return TickAnomalyDecision.NORMAL
+
+    if pct_deviation > incident_threshold:
+        return TickAnomalyDecision.ABNORMAL
+
+    # Anything less and we are absolutely in an incident
+    return TickAnomalyDecision.INCIDENT
```

Comment:
I feel like this reads a little backwards. It might be clearer like:
```
if pct_deviation <= incident_threshold:
    return TickAnomalyDecision.INCIDENT
if pct_deviation <= anomaly_threshold:
    return TickAnomalyDecision.ABNORMAL
return TickAnomalyDecision.NORMAL
```